app_users/views.py

```diff
@@ -103,3 +102,47 @@ def get(self, request, uidb64, token, *args, **kwargs):
         else:
             messages.warning(request, ('The confirmation link was invalid, possibly because it has already been used.'))
             return redirect('homepage')
+
+
+def news_sub(request):
+    form = NewsletterForm()
+    return render(request, 'newsletter.html', {'form': form})
+
+
+def subscribe(request):
+    data = request.GET.get('email')
+    user = User.objects.filter(email__iexact=data)
+    for i in user:
+        if NewsletterSub.objects.get(user=i):
+            NewsletterSub.objects.update(user=i,
+                                         subscribed=True)
+        else:
+            NewsletterSub.objects.create(
+                user=i,
+                subscribed=True
+            )
```

Comment:
Maybe get_or_create could have been used here or maybe .objects.filter(...).update(...) (for the update part).